android-components/components/feature/search/src/test/java/mozilla/components/feature/search/middleware/SearchMiddlewareTest.kt

```diff
@@ -1243,6 +1243,50 @@ class SearchMiddlewareTest {
         assertFalse(store.state.search.disabledSearchEngineIds.contains("bing"))
     }
 
+    @Test
+    fun `WHEN restore hidden search engines action THEN hidden engines are added back to bundled engines list`() = runTestOnMain {
+        val metadataStorage = SearchMetadataStorage(testContext, preferences = lazy { FakeSharedPreferences() }) // SearchMetadataStorage(testContext, preferences = lazy { FakeSharedPreferences() })
+        val searchMiddleware = SearchMiddleware(
+            testContext,
+            ioDispatcher = dispatcher,
+            customStorage = CustomSearchEngineStorage(testContext, dispatcher),
+            metadataStorage = metadataStorage,
+        )
+        val store = BrowserStore(middleware = listOf(searchMiddleware))
+
+        store.dispatch(
+            SearchAction.SetRegionAction(
+                RegionState("US", "US"),
+            ),
+        ).joinBlocking()
```

Comment:
Why are we setting the region in this test?

Reply:
to populate `store.state.search.regionSearchEngines`! it's a common strategy throughout the `SearchMiddlewareTest` test.